blockchain/miner.py

```diff
@@ -25,6 +25,10 @@ def proof_of_work(last_proof):
     print("Searching for next proof")
     proof = 0
     #  TODO: Your code here
+    last_hash = hashlib.sha256(f"{last_proof}".encode()).hexdigest()
+
+    while not valid_proof(last_hash, proof):
+        proof += random.random()
```

Comment:
In addition to floats, you could have also considered using negative numbers and decimal formatting.